apis/CommentsEndpoint.es6.js

```diff
@@ -53,16 +53,14 @@ const parseGetBody = (apiResponse, hasChildren) => {
       });
     }
 
-    comments = parseCommentList(body[1].data.children);
+    comments = body[1].data.children.map(parseCommentData);
   } else if (body.json && body.json.data) {
-    if (hasChildren) { // treeify 'load more comments' replies
-      comments = treeifyComments(parseCommentList(body.json.data.things));
-    } else {
-      comments = parseCommentList(body.json.data.things);
-    }
+
+    const { things } = body.json.data;
+    comments = treeifyComments(things.map(parseCommentData));
   }
 
-  normalizeCommentReplies(comments, (commentJSON, isTopLevel) => {
+  normalizeCommentReplies(comments, true, (commentJSON, isTopLevel) => {
```

Comment:
🔕  Minor style note, just a preference but in no way a blocker for this patch.  For recursive functions like this, I've always been a fan of having a helper that makes the first call with the correct starting value (in this case, that topLevel is `true`). Then you don't have to wonder what this unnamed `true` is, and you can't mess up calling it. We could still export the `_normalizeCommentReplies` function if we desire being able to call it manually.